MCLOUD-6708: Add warning if MAGE_MODE was set as not production (#771)
magento/ece-tools: dist/error-codes.md

| 2023 | install-update:split-db | Enabling a split database will be skipped. |  |
| 2024 | install-update:split-db | The SPLIT_DB variable is missing the configuration for split connection types. |  |
| 2025 | install-update:split-db | Slave connection not set. |  |
| 2026 | pre-deploy:restore-writable-dirs | Failed to restore some data generated during the build phase to the mounted directories | Check the `cloud.log` for more information. |

> | 2027 | validate-config:mage-mode-variable | Environment variable MAGE_MODE has value with not supported mode  | Remove MAGE_MODE environment variable or change its value to "production". Magento Cloud supports only "production" mode. |

Reviewer: "with unsupported" rather than "with not supported" - Clearer English

Author: Thanks Barny.
All texts were updated after your and @meker12 review.

Reviewer: Texts in other files don't seem to have been updated?

> | 2027 | validate-config:mage-mode-variable | Environment variable MAGE_MODE has value with not supported mode  | Remove MAGE_MODE environment variable or change its value to "production". Magento Cloud supports only "production" mode. |

Reviewer 2: ```suggestion
| 2027 | validate-config:mage-mode-variable | Mode value for MAGE_MODE environment variable not supported | Remove MAGE_MODE environment variable, or change its value to "production". Magento Cloud supports only "production" mode. |
```
